driver/driver_test.go

```diff
@@ -123,6 +131,41 @@ var _ = Describe("Contrail Network Driver", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	Specify("creating vswitch for forwarding extension works", func() {
+		err := contrailDriver.StartServing()
+		Expect(err).ToNot(HaveOccurred())
+
+		network, err := hns.GetHNSNetworkByName(common.RootNetworkName)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(network).ToNot(BeNil())
+
+		Expect(network.Name).To(Equal(common.RootNetworkName))
+
+		By("root network is not deleted upon shutdown of driver")
+		err = contrailDriver.StopServing()
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = hns.GetHNSNetworkByName(common.RootNetworkName)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("if root network exists upon driver startup, additional one is not created")
+		netsBefore, err := hns.ListHNSNetworks()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = contrailDriver.StartServing()
+		Expect(err).ToNot(HaveOccurred())
+		_, err = hns.GetHNSNetworkByName(common.RootNetworkName)
+		Expect(err).ToNot(HaveOccurred())
+
+		netsAfter, err := hns.ListHNSNetworks()
+		Expect(err).ToNot(HaveOccurred())
+
```

Comment:
I'm a little confused with that test. What are you testing here? If I correctly understand:
1) In line 138 you created HNSNetwork. 
2) Comparing name in line 142
3) StoppingServing and trying check if Network still exist (line 148)
4) Getting list of networks, starting serving again, and comparing with new list list of networks. 

I don't understand step 4. Why are you not getting list of network before StartServing ?  Only after that you will be sure that network has been created. Also I think that you should invoke GetHNSNetworkByName before start to be sure that StartServing will create new network.